google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -999,6 +1008,26 @@ public Builder setHost(String host) {
       return this;
     }
 
+    /**
+     * Sets the preference for gRPC-GCP extension (default: false). When enabled the gRPC-GCP
+     * channel pool will be used.
+     */
+    public Builder setUseGrpcGcpExtension(boolean useGrpcGcpExtension) {
```

Comment:
It seems that if the user wants to use the gRPC GCP extension they would have to set this as `true` AND configure the options below. Is that true?

If so, I would prefer if we could flatten the option setting by having something like `.withGrpcGcpOptions(GcpManagedChannelOptions)` that automatically sets any flag needed to `true` and simultaneously set the options.

Reply:
Agree. I'll add that.

Reply:
Changed `setGrpcGcpOptions` to `withGrpcGcpOptions`. But I also keep `setUseGrpcGcpExtension` to be able to disable the GCP extension.

Reply:
I think this still will be a little confusing for users, and that many will think that you need to call both to enable it. Do we really need this? Or would it maybe be better to change it to something like `disableGrpcGcpExtension()`?

Reply:
Another point for keeping this is that it is not necessary to provide options to enable the extension - in this case the extension will be enabled with the defaults. I can change this into two methods enable/disable GrpcGcpExtension. What's more common for SpannerOptions?

Reply:
Ah ok, so we have default options that can be applied. Is this case I would overload the `enable` method and have it as:

```java
// Uses the default options
enableGrpcGcpExtension();

// Uses the custom options
enableGrpcGcpExtension(GcpManagedChannelOptions);
```

Reply:
Sounds good! Done.